core/platform/email/mailgun_email_services.py

```diff
@@ -16,10 +16,32 @@
 
 """Provides mailgun api to send email."""
 
+import base64
+import urllib
+import urllib2
 from core.platform.email import gae_email_services
 import feconf
 
-import requests
+
+def post_to_mailgun(data):
+    """Send POST HTTP request to mailgun api. This method is adopted from
+    `requests`'s post method.
+
+    Args:
+        - data: dict. The data to be sent in the request's body.
+
+    Returns:
+         a file-like object.
+         https://docs.python.org/2/library/urllib2.html
+    """
+    auth_str = 'Basic ' + base64.b64encode(
```

Comment:
We typically go with interpolation in this codebase, so let's do 'Basic %s' % ...

Reply:
Done in commit `Move mailgun api check and mailgun domain name check` (Sorry this should have been made to a separate commit.)